bokeh/models/annotations.py

```diff
@@ -543,7 +556,10 @@ class Label(TextAnnotation):
 
     x = Float(help="""
     The x-coordinate in screen coordinates to locate the text anchors.
-    """)
+
+    Datetime values are also accepted, but note that they are immediately
+    converted to milliseconds-since-epoch.
+    """).accepts(Datetime, convert_datetime_type)
```

Comment:
Note that the coords for label had to have an `accepts` added explicitly since they are not number specs to begin with. Perhaps should have/should change label to be vectorized like other annotations.